tutorials/week5-unit-testing.md

```diff
@@ -0,0 +1,727 @@
+
+![Verify Installation](./assets/week5-unit-testing/vscode-jest/verify-installation.jpg)
+
+If restarting Jest does not work, you can run the following in your VSCode by opening VSCode's command palette (`Ctrl + Shift + P` in Windows, `Cmd + Shift + P` in macOS) and run the following - "_Jest: Start All Runners_".
+
+![Manually start Jest Runner](./assets/week5-unit-testing/vscode-jest/manual-jest-runner-start.jpg)
+
+## Configuration
+
+As of the latest stable version, the extension is set to run all tests whenever any change is made to the files containing the tests. You can modify the behaviour to manual execution where you can run specific test cases as and when required. This can be modified by changing the `"jest.autoRun"` setting as follows -
+
+- `"jest.autoRun": {"watch": true}` => will start the jest with the watch flag and leave all tests at "unknown" state until changes are detected.
+- `"jest.autoRun": {"watch": true, "onStartup": ["all-tests"]}` => will start running all tests upon project launch to update overall project test stats, followed by the jest watch for changes.
+- `"jest.autoRun": "off"` => there will be no automatic test run, users will trigger test run by either command or context-menu.
+- `"jest.autoRun": {"watch": false, "onSave": "test-file"}` => the extension will trigger test run for the given test file upon save.
+- `"jest.autoRun": {"watch": false, "onSave": "test-src-file"}` => the extension will trigger test run for the given test or source file upon save.
+
+An example of changing the extension to run in manual mode is demonstrated below -
+
+![Enable Manual Mode](./assets/week5-unit-testing/vscode-jest/disable-auto-run.gif)
+
+> NOTE - You will only be able to `Run` specific test cases as and when required only when `autoRun` is `off`. However, you'll be able to `Debug` any test cases in all available modes.
+
+ 
+## Debugging Tests
+Once the extension is setup and confirmed to be working, we can now begin debugging the tests. You can do so by opening the testing tab from the left sidebar in VSCode to view all the tests present in the current project.
+
+You can also navigate to the specific test file manually, and you will notice some icons/action buttons in the editor next to each test. Right clicking on the icons will bring up a test-specific context menu which will have options to debug the specific test. Be sure to place breakpoints in the specific pieces of code by clicking to the left of the line numbers.
+
+![Debug Test](./assets/week5-unit-testing/vscode-jest/start-debug-test.jpg)
+
+If everything is setup correctly, the debugger in VSCode will pause at the breakpoint you set and you can check the values of the variables by hovering over the said variable. 
+
+![Debugging in Action](./assets/week5-unit-testing/vscode-jest/debugging-tests.jpg)
+
+
+
+
+# General Guidelines For Writing Tests
+
+_Note:_ This will not be used as a reference when grading assignments.
```

Comment:
Would you please also add the 4 points for "good tests" from lesson 5.2, which *will* be what the students are graded on?

Reply:
Added the 4 points for "good tests" from module 5.2.